go/arrow/array/interval.go

```diff
@@ -279,6 +341,28 @@ func (a *DayTimeInterval) setData(data *Data) {
 	}
 }
 
+func (a *DayTimeInterval) getOneForMarshal(i int) interface{} {
+	if a.IsValid(i) {
+		return a.values[i]
+	}
+	return nil
+}
+
+func (a *DayTimeInterval) MarshalJSON() ([]byte, error) {
+	if a.NullN() == 0 {
+		return json.Marshal(a.values)
```

Comment:
same comment, as months, consider making a formatted string that can be reparsed?  I guess whether we use a formatted string or the raw values comes down to who we expect to consume the JSON.  I think for machines the numbers might be more natural, but for human consumption formatted string makes it clearer what the types are.

Reply:
Currently DayTime interval would get marshaled as `{ "days": #, "milliseconds": # }`, the same would be true for month day nano interval type being marshalled as `{ "months": #, "days": #, "nanoseconds": # }`. The only one that currently wouldn't get marshalled into a formatted object or string is the month interval. The intention I had was that the actual types should be defined externally before attempting to parse the JSON as I didn't build any arrow type inference into the JSON reader. It requires being passed a schema or data type.

Reply:
SGTM, might be worth documenting what the value looks like here.

Reply:
Just to add on since my last comment, I've since modified the Month Interval to marshal as a string like you suggested as "1months" etc. but sure, i'll add a specific comment on these marshal functions about their format so it's not only on the top level "to json" functions.

Reply:
comments added.